File: src/main/java/com/aquehora/entities/Usuario.java
package com.aquehora.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity  // Esta clase se comportara como entidad, hara que tu entidad sea persistente en la bd.
@Table(name="usuarios",
        uniqueConstraints = {
            @UniqueConstraint(name="usuario_name_unique",columnNames="name")
        })
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Usuario
{
    @Id
    @SequenceGenerator(name="usuario_sequence",
            sequenceName = "usuario_sequence",
            allocationSize=1)
    @GeneratedValue(strategy = GenerationType.SEQUENCE,
            generator="usuario_sequence")

    @Column(name="id",updatable = false)
    private Long id;

    @Column(name="name",nullable = false,columnDefinition = "TEXT")
    private String nombre;

    @Column(name="correo",nullable = false,columnDefinition = "TEXT")
    private String correo;

    @Column(name="contrasena",nullable = false,columnDefinition = "TEXT")
    private String contrasena;

    @OneToMany(
            mappedBy = "usuario",
            cascade = {CascadeType.PERSIST, CascadeType.REMOVE},
            fetch = FetchType.LAZY
    )
    private List<Nota> notes=new ArrayList<>();
}

File: Assets/Scripts/Entity/Usuario.cs
public class Usuario
{
    public int idUsuario;
    public string nombre;
    public string correo;
    public string contrasena;
    public int fechaDeNacimiento;

    public Usuario()
    {
        idUsuario = -1;
        nombre = "nulo";
        correo = "nulo";
        contrasena = "nulo";
        fechaDeNacimiento = 0;
    }

    public Usuario(int _id, string _nombre, string _correo, string _contrasena, int _fecha)
    {
        idUsuario = _id;
        nombre = _nombre;
        correo = _correo;
        contrasena = _contrasena;
        fechaDeNacimiento = _fecha;
    }
}
